src/app/clothes/page.tsx

```diff
@@ -0,0 +1,87 @@
+'use client';
+import { useState, useEffect } from 'react';
+import { Button } from '@/components/ButtonStyle';
+import { getCategory } from '@/functions/getCategory';
+import { getData } from '@/functions/getData';
+import { normalizeImage } from '@/functions/normalizeImage';
+import { stringifyUrl } from '@/functions/stringifyUrl';
+import { ItemClothes } from '@/types/itemClothes';
+import { Card, CardFooter, Image } from '@nextui-org/react';
+import Link from 'next/link';
+import React from 'react';
+import { dataClothes } from '@/mocks';
+
+const Page = () => {
+  const [data, setData] = useState<ItemClothes[]>([]);
+  const [selectedCategory, setSelectedCategory] = useState<string>('');
+
+  useEffect(() => {
+    const fetchData = async () => {
+      const fetchedData: ItemClothes[] = await getData();
+      setData(fetchedData);
+    };
+
+    fetchData();
+  }, []);
+
+  const filteredData = selectedCategory
+    ? data.filter(item => item.category === selectedCategory)
+    : data;
```

Comment:
@samuelrms this can be wrapped within a useMemo to prevent your variable to be rendered when every fragment of the page gets rehydrated.

Reply:
@vaporwavie useMemo added, according to the tip!